assets/queries/openAPI/general/response_operations_body_schema_undefined/query.rego

```diff
@@ -5,76 +5,92 @@ import data.generic.openapi as openapi_lib
 
 CxPolicy[result] {
 	doc := input.document[i]
-	openapi_lib.check_openapi(doc) != "undefined"
+	version := openapi_lib.check_openapi(doc)
+	version == "2.0"
 
-	operation := doc.paths[path][op]
+	operation := doc.paths[p][op]
+	acceptable_response(operation, op)
 
-	operation_should_have_content := ["get", "put", "post", "delete", "options", "patch", "trace"]
-	common_lib.equalsOrInArray(operation_should_have_content, lower(op))
+	response := operation.responses[code]
+	object.get(response, "schema", "undefined") == "undefined"
 
-	response_code_should_not_have_content := ["204", "304"]
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("paths.%s.%s.responses.%s", [p, op, code]),
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": sprintf("paths.%s.%s.responses.%s.schema is defined", [p, op, code]),
+		"keyActualValue": sprintf("paths.%s.%s.responses.%s.schema is undefined", [p, op, code]),
+		"overrideKey": "2.0",
+	}
+}
 
-	response := operation.responses[code]
-	not common_lib.equalsOrInArray(response_code_should_not_have_content, lower(code))
+CxPolicy[result] {
```

Comment:
I think this two rules can be merged, with the following snippet:
```rego
key := get_key_by_version(version) # should return "schema" for 2.0 and "content" for 3.0
object.get(response, key, "undefined") == "undefined"
```
And chenge version check to `version != "undefined"`...